pytorch/containers/hp_tuning/task.py

```diff
@@ -0,0 +1,159 @@
+    optimizer = optim.SGD(net.parameters(), lr=args.lr,
+                          momentum=args.momentum, nesterov=False)
+
+    # Train / Test the model
+    latest_accuracy = 0.0
+    for epoch in range(1, args.epochs + 1):
+        train(net, train_loader, optimizer)
+        latest_accuracy = test(net, test_loader)
+
+    # The default name of the metric is training/hptuning/metric.
+    # We recommend that you assign a custom name. The only functional
+    # difference is that if you use a custom name, you must set the
+    # hyperparameterMetricTag value in the HyperparameterSpec object in your
+    # job request to match your chosen name.
+    # https://cloud.google.com/ml-engine/reference/rest/v1/projects.jobs#HyperparameterSpec
+    hpt = hypertune.HyperTune()
+    hpt.report_hyperparameter_tuning_metric(
+        hyperparameter_metric_tag='my_metric_tag',
+        metric_value=latest_accuracy,
+        global_step=args.epochs)
+
+    if args.model_name:
+        # Export the trained model
+        torch.save(net.state_dict(), args.model_name)
+
+        if args.job_dir:
+            # Save the model to GCS
+            data_utils.save_model(args.job_dir, args.model_name)
+    else:
+        print('Accuracy: {:.0f}%'.format(latest_accuracy))
+
+
+def main():
+    # Training settings
+    parser = argparse.ArgumentParser(description='PyTorch Sonar Example')
+    parser.add_argument('--job-dir',  # handled automatically by ML Engine
+                        help='GCS location to write checkpoints and export ' \
+                             'models')
+    parser.add_argument('--model-name',
+                        type=str,
+                        default="hptuning_sonar_model",
+                        help='What to name the saved model file')
+    parser.add_argument('--batch-size',
+                        type=int,
+                        default=4,
+                        help='input batch size for training (default: 4)')
+    parser.add_argument('--test-split',
+                        type=float,
+                        default=0.2,
+                        help='split size for training / testing dataset')
+    parser.add_argument('--epochs',
+                        type=int,
+                        default=10,
+                        help='number of epochs to train (default: 10)')
+    parser.add_argument('--lr',  # Specified in the config file
+                        type=float,
+                        default=0.01,
+                        help='learning rate (default: 0.01)')
+    parser.add_argument('--momentum',  # Specified in the config file
+                        type=float,
+                        default=0.5,
+                        help='SGD momentum (default: 0.5)')
+    parser.add_argument('--seed',
+                        type=int,
+                        default=42,
+                        help='random seed (default: 42)')
+    args = parser.parse_args()
+
+    task(args)
```

Comment:
nit: function name should be more descriptive (Example: train_model)
http://google.github.io/styleguide/pyguide.html